pymc4/random_variables/random_variable.py

```diff
@@ -146,15 +146,18 @@ def as_tensor(self):
 class ContinuousRV(RandomVariable):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._transformed_distribution = tfd.TransformedDistribution(
-            distribution=self._distribution, bijector=bijectors.Identity()
-        )
+
 
     def log_prob(self):
         """Log probability computation.
 
         Done based on the transformed distribution, not the base distribution.
         """
+        if not self._transformed_distribution:
```

Comment:
In reference to our talk today about when to instantiate a transformed distribution, @ericmjl and I realized there's three cases

1. Always instantiate transformed distributions on RV Instantiation.
  Pros: There's always a transformed distribution.
  Cons: Additional overhead for RV instantiation

2. Instantiate transformed dist only in InferenceContext
  Pros: Saves on overhead in contexts when people probably don't want logps
  Cons: If you do want logp in forward or freeforward context its unavailable
  https://github.com/pymc-devs/pymc4/pull/83/commits/6f691683f3d65db8c3a487d75a5c0a50c2623a04#diff-720303c73bd15e45708184a55f5a067eR187

3. Only instantiate transformed dist when log_prob is called
  Pros: Transformed dist only there when its needed
  Cons: Additional conditional overhead when calling log_p, from if else or try except block
https://github.com/pymc-devs/pymc4/pull/83/commits/6f691683f3d65db8c3a487d75a5c0a50c2623a04#diff-720303c73bd15e45708184a55f5a067eR156
 
Any thoughts?

In regards to quantitative measurements, in the absence of an inference engine we approximated "inference" by calling logp a 1000 times. Results pending
https://github.com/canyon289/pymc4/blob/always_instantiate/pymc4/tests/test_transformed_speed.py